### ISSUE TITLE
redefined-outer-name when importing subpackage of package which is already imported in outer scope

### ISSUE THREAD

**Issue by jackriccomini** [PyCQA/pylint#2589]
**I don't know if this is intended behaviour or not.** I understand that importing inside a function is not good practice, but it is necessary in the codebase I am working on in order to avoid import loops while some import refactoring takes place. This warning doesn't seem useful or accurate to me, although it seems reasonable to flag function-level imports with a different warning. [This](https://github.com/jackriccomini/pylint/commit/5b880015e804380f5d1011ae171edd35f764be55) change seems to give the behaviour I want.

### Steps to reproduce
1. Create `test.py`
```
import selenium

def foo():
    import selenium.webdriver

foo()
```
### Current behavior
Pylint output includes `Redefining name 'selenium' from outer scope (line 1) (redefined-outer-name)`

### Expected behavior

Pylint output to not include `Redefining name 'selenium' from outer scope (line 1) (redefined-outer-name)`

### pylint --version output
```
No config file found, using default configuration
pylint 1.9.3, 
astroid 1.6.5
Python 2.7.15rc1 (default, Apr 15 2018, 21:51:34) 
[GCC 7.3.0]
```
and
```
pylint 2.2.0-dev
astroid 2.0.4
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0]
```
(behaviour is present on both versions)

**Comment by PCManticore** [comment]
Right, this is definitely not useful. The check was intended for a similar example:
```
some_variable ...

def foo(some_variable):
   # ...

```

**Comment by PCManticore** [comment]
Thanks for creating an issue by the way!
